### DISCUSSION TITLE
created at and updated at colums

### DISCUSSION THREAD

**Discussion by iamchathu** [mikro-orm/mikro-orm#2174] [Q&A]
I'm from `TypeORM` background.

In `TypeORM` we have

```ts
 @CreateDateColumn({ name: 'table_name_created_at' })
  createdAt!: Date;

 @UpdateDateColumn({ name: 'table_name_updated_at' })
  updatedAt!: Date;
```

Which creates 

```sql
`table_name_created_at` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6), `table_name_updated_at` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6),
```

What will be the equviant of it in `mickro-orm`?

**Comment by B4nan** [comment]
There are no special decorators for this, so if the DDL you provided is exactly what you want:

```ts
@Property({ length: 6, defaultRaw: 'CURRENT_TIMESTAMP(6)' })
createdAt!: Date;

@Property({ length: 6, defaultRaw: 'CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6)' })
updatedAt!: Date;
```

The MikroORM way would be to use regular property initializers, so you have the values right at the moment when the entity instance is created. Note that for that you can't have 6 digits precision, as JS can't handle that with native `Date` object (you'd have to use some custom type to map it differently than to a `Date` object):

```ts
@Property({ length: 3 })
createdAt: Date = new Date();

@Property({ length: 3, onUpdate: () => new Date() })
updatedAt: Date = new Date();
```

(you can also combine those approaches)
